graphql/batch_executor.go

```diff
@@ -157,10 +158,14 @@ func executeBatchWorkUnit(unit *WorkUnit) []*WorkUnit {
 func executeNonExpensiveWorkUnit(unit *WorkUnit) []*WorkUnit {
 	results := make([]interface{}, 0, len(unit.sources))
 	for idx, src := range unit.sources {
-		if unit.objectName == "Query" {
-			unit.Ctx = context.WithValue(unit.Ctx, nonExpensive{}, struct{}{})
+		ctx := unit.Ctx
+
+		// ObjectName is initially either "query" or "mutation" for the initial work unit, and for later work units
+		// is the parents object name. We only want to exclude mutations on the initial work unit level.
```

Comment:
```
Fields on the Mutation object should not be marked as "non-Expensive" because they are guaranteed to only execute once.  The only fields we want to validate "expensiveness" on are non-Mutation Fields.
```